addons/mail/static/src/model/record_list.js

```diff
@@ -1,74 +1,252 @@
-import { reactive, toRaw } from "@odoo/owl";
-import { isRecord } from "./misc";
+import { markRaw, reactive, toRaw } from "@odoo/owl";
+import { IS_RECORD_LIST_SYM, isRecord } from "./misc";
 
 /** @param {RecordList} reclist */
 function getInverse(reclist) {
-    return reclist.owner.Model._.fieldsInverse.get(reclist.name);
+    return reclist._.owner.Model._.fieldsInverse.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function getTargetModel(reclist) {
-    return reclist.owner.Model._.fieldsTargetModel.get(reclist.name);
+    return reclist._.owner.Model._.fieldsTargetModel.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function isComputeField(reclist) {
-    return reclist.owner.Model._.fieldsCompute.get(reclist.name);
+    return reclist._.owner.Model._.fieldsCompute.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function isSortField(reclist) {
-    return reclist.owner.Model._.fieldsSort.get(reclist.name);
+    return reclist._.owner.Model._.fieldsSort.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function isEager(reclist) {
-    return reclist.owner.Model._.fieldsEager.get(reclist.name);
+    return reclist._.owner.Model._.fieldsEager.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function setComputeInNeed(reclist) {
-    reclist.owner._.fieldsComputeInNeed.set(reclist.name, true);
+    reclist._.owner._.fieldsComputeInNeed.set(reclist._.name, true);
 }
 
 /** @param {RecordList} reclist */
 function setSortInNeed(reclist) {
-    reclist.owner._.fieldsSortInNeed.set(reclist.name, true);
+    reclist._.owner._.fieldsSortInNeed.set(reclist._.name, true);
 }
 
 /** @param {RecordList} reclist */
 function isComputeOnNeed(reclist) {
-    return reclist.owner._.fieldsComputeOnNeed.get(reclist.name);
+    return reclist._.owner._.fieldsComputeOnNeed.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function isSortOnNeed(reclist) {
-    return reclist.owner._.fieldsSortOnNeed.get(reclist.name);
+    return reclist._.owner._.fieldsSortOnNeed.get(reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function computeField(reclist) {
-    reclist.owner._.compute(reclist.owner, reclist.name);
+    reclist._.owner._.compute(reclist._.owner, reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function sortField(reclist) {
-    reclist.owner._.sort(reclist.owner, reclist.name);
+    reclist._.owner._.sort(reclist._.owner, reclist._.name);
 }
 
 /** @param {RecordList} reclist */
 function isOne(reclist) {
-    return reclist.owner.Model._.fieldsOne.get(reclist.name);
+    return reclist._.owner.Model._.fieldsOne.get(reclist._.name);
 }
 
-/** * @template {Record} R */
-export class RecordList extends Array {
-    /** @type {import("./record").Record} */
-    owner;
+export class RecordListInternal {
```

Comment:
No separate file for this one?

Reply:
`RecordListInternal` uses some alias function and it would be strange to move them to `RecordListInternal` file.

IMO it doesn't really matter whether the "internal" parts are in a dedicated file or not. In general it's good to put each classes in their own file, but here it's a bit more practical to have it in the same file